### ISSUE TITLE
Region Role and Section - Learn Accessibility by Building a Quiz

### ISSUE THREAD

**Issue by alfonsusac** [freeCodeCamp/freeCodeCamp#47451]
### Describe the Issue

In Step 16 the tutorial told us to use both section and `role="region"`.
According to [MDN Web Docs about Region Role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role) states that

> Using the [`<section>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section) element will automatically communicate a section has a role of `region` if it is given an accessible name. Developers should always prefer using the correct semantic HTML element, in this case `<section>` , over using ARIA.

And in [MDN Web Docs \<section\> element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section) also states that for best practice there should be a heading inside the section element for accessibility purposes.

I am now left wondering why do we still need to declare the role and `aria-labelledby` in the section element in this step?

Thank you

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-16

### Your code

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta name="description" content="freeCodeCamp Accessibility Quiz practice project" />
    <title>Accessibility Quiz</title>
    <link rel="stylesheet" href="styles.css" />
  </head>
  <body>
    <header>
      <img id="logo" src="https://cdn.freecodecamp.org/platform/universal/fcc_primary.svg">
      <h1>HTML/CSS Quiz</h1>
      <nav>
        <ul>
          <li><a>INFO</a></li>
          <li><a>HTML</a></li>
          <li><a>CSS</a></li>
			  </ul>
      </nav>
    </header>
    <main>
      <form method="post" action="https://freecodecamp.org/practice-project/accessibility-quiz">
        <section role="region" aria-labelledby="student-info">
          <h2 id="student-info">Student Info</h2>
        </section>
        <section role="region" aria-labelledby="html-questions">
          <h2 id="html-questions">HTML</h2>
        </section>
        <section role="region" aria-labelledby="css-questions">
          <h2 id="css-questions">CSS</h2>
        </section>
      </form>
    </main>
  </body>
</html>

```

### Expected behavior

To not use roles in section unless its a div and we're building a semantic that is not defined in the HTML element?

### System

- Device: PC
- OS: Windows 10
- Browser: Brave
- Version: Latest

**Comment by ShaunSHamilton** [comment]
In general, we recommend bringing your questions to the forum where more people will see it, and it is more conducive for this kind of discussion.

---


> I am now left wondering why do we still need to declare the role and aria-labelledby in the section element in this step?

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role#description

> And in [MDN Web Docs <section> element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section) also states that for best practice there should be a heading inside the section element for accessibility purposes.

The linked lesson teaches you to add `h2` elements to the `section` elements.

**Comment by bbsmooth** [comment]
I responded to this in the [associated forum post](https://forum.freecodecamp.org/t/region-role-and-section-learn-accessibility-by-building-a-quiz/550755), but just to summarize here, in the real world, you would probably not add `role="region"` to these `section` elements since giving them an accessible name with `aria-labelledby` automatically gives them the role of "region" and the general rule is not to add attributes like this if they aren't needed. But having `role="region"` on them is not causing any accessibility issues, it's just needless duplication. 

The real danger here as I see it is that these steps could be taken to imply that every `section` element should have a role of "region" and an accessible name, which is definitely not the case. Even if a section has a heading associated with it, that does not necessarily require you to turn the section into a landmark area by giving it an accessible name via `aria-labelledby'. In fact, you don't necessarily want to make every `section` a landmark region because too many landmarks is just as unhelpful as not having any landmarks. It is possible to pollute the landmark space to the point that it basically becomes meaningless to screen reader users. So sections should only be turned into landmarks if they are significant enough to warrant it. And to be honest, for this small quiz I would say that the sections should not be turned into landmarks.

But I understand that the point of this is to teach people about `role` attribute. It's just unfortunate that current example being used is not a great one. But it's not technically inaccessible. Whether we want to rework this course to use a different example I'll let others decide.

**Comment by bbsmooth** [comment]
If you want to turn these sections into landmarks then using a `section` element with `aria-labelledby` is the best semantic option. You just don't need to add `role="region"`.

**Comment by naomi-lgbt** [comment]
This has sat here for a while. Going to open it up for contributions. A PR which resolves this issue should implement sidemt's suggestion above.

**Comment by moT01** [comment]
It would be nice to make some adjustments here, but this issue didn't get much traction for a year and a half. I think it's safe to close this as stale. If anyone objects, feel free to leave a comment. Thanks and happy coding 🎉
